From 921d258ac4ae21ac8c752b8bc119ab98d990a4c8 Mon Sep 17 00:00:00 2001
From: Davidson K <davidson.kumaresan@arm.com>
Date: Mon, 18 Jul 2022 19:27:20 +0530
Subject: [PATCH 08/14] do not use init_info

init_info is a pointer passed from spmc to sp through x0 register.
Hafnium can either pass fdt address or mem size (src/load.c:725).
To pass fdt, the SPMC manifest (tc_spmc_optee_sp_manifest.dts) in
TF-A should have the "fdt_filename". In the current setup, it passes
memsize which should be an integer. Treating that as a pointer in
crypto and internal trusted storage services causes problems. Hence
removing it.

Signed-off-by: Davidson K <davidson.kumaresan@arm.com>
Change-Id: I6e51ed7731fc1e4a474238b8a9fee96a56d4d245

Upstream-Status: Pending [Not submitted to upstream yet]
---
 deployments/crypto/common/crypto_sp.c                | 1 -
 deployments/internal-trusted-storage/common/its_sp.c | 1 -
 2 files changed, 2 deletions(-)

diff --git a/deployments/crypto/common/crypto_sp.c b/deployments/crypto/common/crypto_sp.c
index ce236d9..38bf471 100644
--- a/deployments/crypto/common/crypto_sp.c
+++ b/deployments/crypto/common/crypto_sp.c
@@ -36,7 +36,6 @@ void __noreturn sp_main(struct ffa_init_info *init_info)
 	if (sp_init(&own_id) != 0) goto fatal_error;
 
 	config_ramstore_init();
-	sp_config_load(init_info);
 
 	/* Create a storage backend for persistent key storage - prefer ITS */
 	storage_backend = storage_factory_create(storage_factory_security_class_INTERNAL_TRUSTED);
diff --git a/deployments/internal-trusted-storage/common/its_sp.c b/deployments/internal-trusted-storage/common/its_sp.c
index 18df88f..b6f5694 100644
--- a/deployments/internal-trusted-storage/common/its_sp.c
+++ b/deployments/internal-trusted-storage/common/its_sp.c
@@ -30,7 +30,6 @@ void sp_main(struct ffa_init_info *init_info)
 	struct storage_backend *storage_backend;
 
 	/* Boot */
-	(void) init_info;
 
 	ffa_res = ffa_id_get(&own_id);
 	if (ffa_res != FFA_OK) {
-- 
2.34.1

